### ISSUE TITLE
Wayland support

### ISSUE THREAD

**Issue by Elinvention** [SuperTux/supertux#323]
I tried to run the game in a "GNOME on Wayland" session with `SDL_VIDEODRIVER=wayland supertux`, but that caused a segmentation fault.
I'm using SuperTux v0.4.0, GNOME 3.18 and SDL 2.0.2 from Debian Testing.

This is what `SDL_VIDEODRIVER=wayland gdb supertux2` (without debugging symbols) had to say:

```
Starting program: /usr/games/supertux2 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7fffe9a1e700 (LWP 1834)]
[New Thread 0x7fffe4d62700 (LWP 1835)]

Program received signal SIGSEGV, Segmentation fault.
0x00007ffff49370ad in XQueryExtension ()
   from /usr/lib/x86_64-linux-gnu/libX11.so.6
```

**Comment by n3rdopolis** [comment]
It works on Wayland If you compile with glbinding
https://github.com/hpicgs/glbinding
(add  -DGLBINDING_ENABLED=ON  to the cmake command)

Or I think if you use the 
-renderer sdl 
option when calling Supertux

**Comment by Elinvention** [comment]
It works, thank you! Of course it would be better if the game automatically recognized the renderer to use.

**Comment by Elinvention** [comment]
Ops, my bad... actually it doesn't work with `SDL_VIDEODRIVER=wayland supertux2 --renderer sdl`. I have NOT tried with glbinding. It worked because i launched it without the SDL_VIDEODRIVER variable and so it was rendering through XWayland.

**Comment by Elinvention** [comment]
@tobbi @n3rdopolis I tested with glbinding and SDL 2.0.4 too and it doesn't work with `SDL_VIDEODRIVER=wayland`, so the issue remains.

**Comment by tobbi** [comment]
This is weird. If you could figure out more, that would be great. Unfortunately, I have no idea whatsoever about the issues in place, so I don't know what information we'd need to collect in order to get this mitigated / fixed.

**Comment by tobbi** [comment]
When it SIGSEGV's, where exactly does it do that? Can you provide a stacktrace?

**Comment by Elinvention** [comment]
Here is the backtrace from gdb: [gdb.txt](https://github.com/SuperTux/supertux/files/211360/gdb.txt). Can you reproduce the issue?

**Comment by tobbi** [comment]
I haven't tried yet. But it's still good to have the stacktrace here for further reference.

**Comment by tobbi** [comment]
Do you have any possibility to link against SDL 2.0.4 to see if this helps?

From the release log https://www.libsdl.org/tmp/SDL/WhatsNew.txt:
* Enabled building with Mir and Wayland support by default.

I took a look at the stacktrace again and it crashes around here:
https://github.com/SuperTux/supertux/blob/v0.4.0/src/supertux/main.cpp#L257

Ergo, this is an error in the SDL subsystem which we cannot fix. Please, as I mentioned above, try again with 2.0.4.

**Comment by Elinvention** [comment]
I compiled git master version with SDL 2.0.4 and exported SDL_VIDEODRIVER=wayland and I got this results:

- no glbinding, no --renderer sdl
- no glbinding, with --renderer sdl
- with glbinding, no --renderer sdl
- with glbinding and --renderer sdl

All this test cases had the same result: high probability of segmentation fault and low probability of game launching properly. When the game launches properly in windowed mode, there are no window decorations. If I toggle full screen in game options, sometimes the whole window disappears, but the game keeps running and music keeps playing. Of course, if I unset SDL_VIDEODRIVER the game works fine, but via XWayland.

I think if SDL_init fails, there must be something which sometimes is run before that interferes with SDL. Or perhaps something in the compilation... I can say that I compiled Xonotic with SDL2 and it was working on Wayland (except relative mouse input).
